packages/web-app-files/src/composables/scrollTo/useScrollTo.ts

```diff
@@ -22,7 +34,38 @@ export const useScrollTo = (): ScrollToResult => {
     return queryItemAsString(unref(detailsQuery))
   })
 
-  const scrollToResource = (resourceId: Resource['id'], options = { forceScroll: false }) => {
+  const router = useRouter()
+  const route = useRoute()
+  const files = computed((): Array<Resource> => store.getters['Files/files'])
+  const currentRoute = useRouteName()
+  const itemsPerPageQuery = useRouteQuery('items-per-page', '1')
+  const pageQuery = useRouteQuery('page', '1')
+  const currentViewModeQuery = useRouteQuery(
+    `${unref(currentRoute)}-view-mode`,
+    ViewModeConstants.defaultModeName
+  )
+
+  const currentViewMode = computed((): string => queryItemAsString(currentViewModeQuery.value))
+  const viewMode = useViewMode(currentViewMode)
+  const sortFields = computed((): SortField[] => {
+    if (unref(viewMode) === ViewModeConstants.tilesView.name) {
+      return determineResourceTilesSortFields(unref(files)[0])
+    }
+    return determineResourceTableSortFields(unref(files)[0])
+  })
+  const { items: sortedItems } = useSort({ items: files, fields: sortFields })
+
+  const scrollToResource = async (resourceId: Resource['id'], options = { forceScroll: false }) => {
+    if (unref(sortedItems)) {
+      const index = findIndex(unref(sortedItems), (item: Resource) => item.id === resourceId)
```

Comment:
I'd prefer doing this lookup only in the `if (!resourceElement) {...}` and check the "current page DOM" for the fileId first?

Reply:
I wouldn't prefer that tbh 😬

Reply:
I guess I have an idea why, but would you mind elaborating?

Reply:
yep, that's what I meant

Reply:
... but I would solve the whole issue a little differently, wrote it down in detail in my review